# HG changeset patch
# User Rik <rik@octave.org>
# Date 1449938784 28800
#      Sat Dec 12 08:46:24 2015 -0800
# Node ID ac1f8c640d8f31fcf484e2f1b82e1cd5ef558910
# Parent  6774cc464ea074805d042f5010a44df0f9dbdf5b
2015 Code Sprint: colamd.cc: use ovl().

diff --git a/libinterp/dldfcn/colamd.cc b/libinterp/dldfcn/colamd.cc
--- a/libinterp/dldfcn/colamd.cc
+++ b/libinterp/dldfcn/colamd.cc
@@ -283,16 +283,18 @@ Xerox PARC, and @nospell{Esmond Ng}, Oak
 #ifdef HAVE_COLAMD
 
   int nargin = args.length ();
   int spumoni = 0;
 
   if (nargout > 2 || nargin < 1 || nargin > 2)
     print_usage ();
 
+  retval.resize (nargout == 2 ? 2 : 1);
+
   // Get knobs
   OCTAVE_LOCAL_BUFFER (double, knobs, COLAMD_KNOBS);
   COLAMD_NAME (_set_defaults) (knobs);
 
   // Check for user-passed knobs
   if (nargin == 2)
     {
       NDArray User_knobs = args(1).array_value ();
@@ -518,16 +520,18 @@ Xerox PARC, and @nospell{Esmond Ng}, Oak
 #ifdef HAVE_COLAMD
 
   int nargin = args.length ();
   int spumoni = 0;
 
   if (nargout > 2 || nargin < 1 || nargin > 2)
     print_usage ();
 
+  retval.resize (nargin == 2 ? 2 : 1);
+
   // Get knobs
   OCTAVE_LOCAL_BUFFER (double, knobs, COLAMD_KNOBS);
   COLAMD_NAME (_set_defaults) (knobs);
 
   // Check for user-passed knobs
   if (nargin == 2)
     {
       NDArray User_knobs = args(1).array_value ();
@@ -658,16 +662,18 @@ permutations on the tree.\n\
 {
   octave_value_list retval;
 
   int nargin = args.length ();
 
   if (nargout > 2 || nargin < 1 || nargin > 2)
     print_usage ();
 
+  retval.resize (nargout == 2 ? 2 : 1);
+
   octave_idx_type n_row = 0, n_col = 0;
   octave_idx_type *ridx = 0, *cidx = 0;
 
   if (args(0).is_sparse_type ())
     {
       if (args(0).is_complex_type ())
         {
           SparseComplexMatrix scm = args(0).sparse_complex_matrix_value ();
